src/java.desktop/share/classes/com/sun/java/swing/plaf/gtk/GTKFileChooserUI.java

```diff
@@ -432,6 +432,10 @@ public  DoubleClickListener(JList<?> list) {
         public void mouseClicked(MouseEvent e) {
             if (SwingUtilities.isLeftMouseButton(e) && e.getClickCount() == 2) {
                 int index = list.locationToIndex(e.getPoint());
+                File currentDirectory = getFileChooser().getCurrentDirectory();
+                if (currentDirectory.getParentFile() == null && index == 1) {
```

Comment:
Adding a Comment explaining the statement `index == 1` would be better to understand that it signifies root directory index.

Reply:
Added a comment. 
Basically, `getParentFile` method is used to check root directory and `index == 1` is the index of `../` directory.